share/shwap/p2p/shrex/shrex_getter/shrex.go

```diff
@@ -143,105 +95,164 @@ func (sg *Getter) Stop(ctx context.Context) error {
 	return sg.archivalPeerManager.Stop(ctx)
 }
 
-func (sg *Getter) GetSamples(context.Context, *header.ExtendedHeader, []shwap.SampleCoords) ([]shwap.Sample, error) {
-	return nil, fmt.Errorf("getter/shrex: GetShare %w", shwap.ErrOperationNotSupported)
-}
+func (sg *Getter) GetSamples(
+	ctx context.Context,
+	header *header.ExtendedHeader,
+	coords []shwap.SampleCoords,
+) ([]shwap.Sample, error) {
+	// short circuit if the data root is empty
+	if header.DAH.Equals(share.EmptyEDSRoots()) {
+		return []shwap.Sample{}, nil
+	}
 
-func (sg *Getter) GetRow(_ context.Context, _ *header.ExtendedHeader, _ int) (shwap.Row, error) {
-	return shwap.Row{}, fmt.Errorf("getter/shrex: GetRow %w", shwap.ErrOperationNotSupported)
+	var err error
+	ctx, span := tracer.Start(ctx, "shrex/get-samples")
+	defer func() {
+		utils.SetStatusAndEnd(span, err)
+	}()
```

Comment:
There will be issues because you shadow `err` var once and in `sg.executeRequest(ctx, logger, header, request.Name(), req, verify)` call you don't assign returned error to this var